src/Adapter/ImageType/ImageSettingsHelper.php

```diff
@@ -0,0 +1,128 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\ImageType;
+
+use Image;
+use PrestaShop\PrestaShop\Core\ImageType\ImageSettingsHelperInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * Helper functions for Image Settings page.
+ */
+class ImageSettingsHelper implements ImageSettingsHelperInterface
+{
+    /**
+     * @var int
+     */
+    protected $max_execution_time = 7200;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var string
+     */
+    public $productImagesDir;
+
+    /**
+     * @var bool
+     */
+    public $showDuplicatesAlert;
+
+    /**
+     * @param TranslatorInterface $translator
+     * @param string $productImagesDir
+     */
+    public function __construct(TranslatorInterface $translator, string $productImagesDir)
+    {
+        $this->translator = $translator;
+        $this->productImagesDir = $productImagesDir;
+        $this->showDuplicatesAlert = file_exists($productImagesDir . 'duplicates' . DIRECTORY_SEPARATOR);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function moveImagesToNewFileSystem(): array
+    {
+        if (!Image::testFileSystem()) {
+            return [$this->translator->trans(
+                'Error: Your server configuration is not compatible with the new image system. No images were moved.',
+                [],
+                'Admin.Design.Notification'
+            )];
+        }
+
+        ini_set('max_execution_time', (string) $this->max_execution_time);
+        $this->max_execution_time = (int) ini_get('max_execution_time');
```

Comment:
I don't get it.
You ini_set the same value as you retrieve :thinking:

Reply:
I don't understand this myself also, but this is how this was in legacy code, so I think this is necessary.

Reply:
Could you please add acomment about it, I'm afraid if someone like me come on this page, he will remove it :sweat_smile: